### PR TITLE
Add log fields for Go messages

### PR DESCRIPTION
**Pull request by htdvisser**
Currently does not contain tests, will be implemented when #2 is done

### CODE REVIEW

File: logfields/logfields.go [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+	}
+}
+
+func fillString(fields log.Fields, key string, value string) {
+	if value != "" {
+		fields[key] = value
+	}
+}
+
+type hasID interface {
```
Comment (rvolosatovs) [inline logfields/logfields.go]: why this naming format instead of `idGetter` etc. ?(related to all other cases)

File: logfields/logfields.go [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+	fillMessage,
+	fillProtocolMetadata,
+	fillDownlinkOption,
+	fillGatewayConfiguration,
+	fillProtocolConfiguration,
+	fillPayload,
+}
+
+// ForMessage returns the log fields for the given message
+func ForMessage(m interface{}) log.Fields {
```
Comment (rvolosatovs) [inline logfields/logfields.go]: `FromMessage`

Comment (htdvisser) [inline logfields/logfields.go]: We're getting the fields for a message, ergo `logfields.ForMessage()`

File: logfields/logfields.go (hunk 1 above)
Comment (htdvisser) [inline logfields/logfields.go]: Because imo `m.(hasActivationMetadata)` reads better than `m.(activationMetadataGetter)`

**Comment by htdvisser** [comment]
> Performance of type assertions

I think having a working, flexible MVP is more important than performance at the moment. We can always benchmark and optimize this later. I can live with being a couple of nanoseconds slower. Also, this way we save quite some development time, because we don't have to implement this for every field/message (and maintain it if we make changes).

**Pull request merged**